src/components/Personalization/createSubscribeMessageFeed.js

```diff
@@ -27,11 +28,12 @@ const validateOptions = ({ options }) => {
 };
 
 export default ({ collect }) => {
-  let subscriptionHandler;
+  const subscription = createSubscription();
   let surfaceIdentifier;
   const run = ({ surface, callback }) => {
-    subscriptionHandler = callback;
+    const unsubscribe = subscription.add(callback);
     surfaceIdentifier = surface;
```

Comment:
Same problem here. You are overwriting the surface instead of having one for each call.